### PR TITLE
Adding a bunch of blocks.

### PR DESCRIPTION
**Pull request by tristanmorgan**
Added new Cactus, (mossy) cobblestone_stairs (Smooth) quartz stairs and stone brick stairs.
tweaks to sand to allow cactus and deadbush(s). Leaves drop sticks, Kelp doesn't liquid drop and Sand can grow cactus (cacti?).

I think the lists of addItem and regBlock in register.go should probably get sorted someday.

### CODE REVIEW

File: server/block/cactus.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+	// Age is the groth state of cactus. values from 0 to 15 maybe?
+	Age int
+}
+
+// FlammabilityInfo ...
+func (c Cactus) FlammabilityInfo() FlammabilityInfo {
+	return newFlammabilityInfo(5, 5, true)
+}
+
```
Comment (DaPigGuy) [inline server/block/cactus.go:23]:
```suggestion
```

Cactus blocks are not flammable

File: server/block/cactus.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+func (c Cactus) RandomTick(pos cube.Pos, w *world.World, r *rand.Rand) {
+	if c.Age < 15 && r.Float64() < 0.01 {
+		c.Age++
+		w.SetBlock(pos, Cactus{Age: c.Age + 1}, nil)
+	}
+}
+
+// BreakInfo ...
+func (c Cactus) BreakInfo() BreakInfo {
+	return newBreakInfo(2, alwaysHarvestable, axeEffective, oneOf(c))
```
Comment (DaPigGuy) [inline server/block/cactus.go:50]:
```suggestion
	return newBreakInfo(0.4, alwaysHarvestable, nothingEffective, oneOf(c))
```

File: server/block/cactus.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/particle"
+	"math/rand"
+)
+
+// A Cactus is a naturally occurring block found in deserts
+type Cactus struct {
+	solid
+	snare
```
Comment (DaPigGuy) [inline server/block/cactus.go:13]:
```suggestion
```

File: server/block/cactus.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/particle"
+	"math/rand"
+)
+
+// A Cactus is a naturally occurring block found in deserts
+type Cactus struct {
+	solid
```
Comment (DaPigGuy) [inline server/block/cactus.go:12]: Cactus blocks do not appear to have a normal AABB?

File: server/block/cactus.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/block/cube"
+	"github.com/df-mc/dragonfly/server/world"
+	"github.com/df-mc/dragonfly/server/world/particle"
+	"math/rand"
+)
+
+// A Cactus is a naturally occurring block found in deserts
```
Comment (DaPigGuy) [inline server/block/cactus.go:10]: Comment should begin with `// Cactus` and have proper punctuation

File: server/block/cactus.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+type Cactus struct {
+	solid
+	snare
+
+	// Age is the groth state of cactus. values from 0 to 15 maybe?
```
Comment (DaPigGuy) [inline server/block/cactus.go:15]: The second part of the comment should not be phrased as a question

File: server/block/cactus.go [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+		w.SetBlock(pos, nil, nil)
+		w.AddParticle(pos.Vec3Centre(), particle.BlockBreak{Block: c})
+	}
+}
+
+// RandomTick ...
+func (c Cactus) RandomTick(pos cube.Pos, w *world.World, r *rand.Rand) {
+	if c.Age < 15 && r.Float64() < 0.01 {
+		c.Age++
+		w.SetBlock(pos, Cactus{Age: c.Age + 1}, nil)
```
Comment (DaPigGuy) [inline server/block/cactus.go:44]: There seems to be missing logic for new cactus blocks being created when age reaches 15

**Review by DaPigGuy — CHANGES_REQUESTED** [review]

File: server/block/cactus.go (hunk 7 above)
Comment (JustTalDevelops) [inline server/block/cactus.go:44]: You update the age right above?

File: server/block/cactus.go (hunk 5 above)
Comment (JustTalDevelops) [inline server/block/cactus.go:10]: I would remove the `A ` prefix here to be consistent with the rest of the blocks in the package.

File: server/block/cactus.go (hunk 6 above)
Comment (JustTalDevelops) [inline server/block/cactus.go:15]: They do range from 0 - 15.

File: server/block/cactus.go (hunk 1 above)
Comment (tristanmorgan) [reply server/block/cactus.go:23]: I'll knock this out, but is there a good resource to find the properties like this?

Comment (JustTalDevelops) [reply server/block/cactus.go:23]: The wiki usually has some solid info on this stuff,

**Comment by tristanmorgan** [comment]
Closing this one in favour of smaller more targeted PR's #497 and others.

**Pull request closed without merging**